app/code/Magento/Checkout/view/frontend/web/js/shopping-cart.js

```diff
@@ -5,23 +5,32 @@
 
 define([
     'jquery',
-    'jquery-ui-modules/widget'
-], function ($) {
+    'Magento_Ui/js/modal/confirm',
+    'jquery-ui-modules/widget',
+    'mage/translate'
+], function ($, confirm) {
     'use strict';
 
     $.widget('mage.shoppingCart', {
         /** @inheritdoc */
         _create: function () {
-            var items, i, reload;
+            var items, i, reload, self = this;
 
             $(this.options.emptyCartButton).on('click', $.proxy(function () {
-                $(this.options.emptyCartButton).attr('name', 'update_cart_action_temp');
-                $(this.options.updateCartActionContainer)
-                    .attr('name', 'update_cart_action').attr('value', 'empty_cart');
+                confirm({
```

Comment:
Because this confirmation modal is an added behaviour I can imagine there are cases with the desire to keep the (then) "old" behaviour.

What about making this more developer friendly by refactoring it into two separate functions and easily extendable through mixins? A function which is responsible for handling the `click` event on the `this.options.emptyCartButton` element and one which actually modifies its attributes and submits the parent form.

If you don't want the confirmation modal you could do something like the following. Override the `onClearCart` method to only call `clearCart()` without the confirmation modal.

```
onClearCart: function () {
    this.clearCart();
}
```

Reply:
Updated, can you please review?